packages/js/onboarding/src/components/WooOnboardingTaskReferral/index.ts

```diff
@@ -0,0 +1,71 @@
+type StoredItem< T > = {
+	data: T;
+	expiry: string;
+};
+
+/**
+ * This constructor function exists for type safety purposes, so when you create a pair of set/get functions they have the
+ *  same type for the value they store/retrieve.
+ */
+export function createStorageUtils< T >(
+	key: string,
+	durationInSeconds: number,
+	type: 'session' | 'local' = 'local'
+) {
+	const storage = type === 'session' ? sessionStorage : localStorage;
```

Comment:
Just a nitpick but I think `sessionStorage` is a safer default in general because it is isolated to the current tab. `localStorage` might introduce race conditions between tabs 🤔